src/Microsoft.DocAsCode.Build.Engine/MarkdownProviders/JsonServiceProviders.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.Engine.MarkdownProviders
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Composition;
+
+    using Microsoft.DocAsCode.Dfm;
+    using Microsoft.DocAsCode.MarkdownLite;
+    using Microsoft.DocAsCode.Plugins;
+
+    [Export("json", typeof(IMarkdownServiceProvider))]
+    public class JsonServiceProviders
+    {
+        public IMarkdownService CreateMarkdownService(MarkdownServiceParameters parameters)
+        {
+            return new JsonRenderService(
+                parameters.BasePath,
+                parameters.Tokens,
+                MarkdownTokenTreeValidatorFactory.Combine(TokenTreeValidator));
+        }
+
+        [ImportMany]
+        public IEnumerable<IMarkdownTokenTreeValidator> TokenTreeValidator { get; set; }
+
+        private sealed class JsonRenderService : IMarkdownService
+        {
+            private readonly DfmEngineBuilder _builder;
+
+            private readonly ImmutableDictionary<string, string> _tokens;
+
+            public JsonRenderService(string baseDir, ImmutableDictionary<string, string> tokens, IMarkdownTokenTreeValidator tokenTreeValidator)
+            {
+                _builder = DocfxFlavoredMarked.CreateBuilder(baseDir);
+                _builder.TokenTreeValidator = tokenTreeValidator;
+                _tokens = tokens;
+            }
+
+            public MarkupResult Markup(string src, string path)
+            {
+                var dependency = new HashSet<string>();
+                var json = _builder.CreateDfmEngine(new JsonRenderer() { Tokens = _tokens }).Markup(src, path, dependency);
+                var result = new MarkupResult
+                {
+                    // I think it should be rename 
+                    Html = json,
```

Comment:
or I can just return a string instead of a class like 'MarkupResult'@vwxyzh 

---
In reply to: [74545903](https://github.com/dotnet/docfx/pull/602#discussion_r74545903) [](ancestors = 74545903,74544397)